ENIGMAsystem/SHELL/Platforms/xlib/XLIBmain.cpp

```diff
@@ -464,6 +464,33 @@ void action_end_game() {
   game_end();
 }
 
+void url_open(std::string url,std::string target,std::string options)
+{
+	if (system(NULL))
+	{
+		system(("xdg-open \""+url+"\"").c_str());
```

Comment:
For best results, use single quotes, and escape `url` (do a replace-all for `"'"` → `"'\\''"`). Alternatively, I gave TKG a function that invokes a binary using `execl` in his last pull request; using that will eliminate the need for this escaping.

Reply:
And..... travis is back to randomly segfaulting........

Reply:
@faissaloo I agree with Josh on this, also execl is more secure anyway. I've been reading all over the internet how evil using system() is. Here's the file he's talking about: https://github.com/enigma-dev/enigma-dev/pull/1075/files

Reply:
Alright, I've changed it to use execlp now.